src/main/java/com/joss/conductor/mobile/ConductorConfig.java

```diff
@@ -0,0 +1,379 @@
+        }
+
+        return Paths.get(System.getProperty("user.dir"), path.normalize().toString()).normalize().toString();
+    }
+
+    private void readConfig(InputStream is) {
+        Yaml yaml = new Yaml();
+        Map<String, Object> config = (Map<String, Object>)yaml.load(is);
+
+        String environmentPlatformName = System.getProperty("conductorPlatformName");
+        if(environmentPlatformName != null) {
+            platformName = Platform.valueOf(environmentPlatformName);
+        } else if(config.containsKey("platformName")) {
+            platformName = Platform.valueOf((String)config.get("platformName"));
+        }
+
+        Map<String, Object> defaults = (Map<String, Object>)config.get("defaults");
+        if(defaults != null) {
+            readProperties(defaults);
+
+            Map<String, Object> platformDefaults = null;
+            switch (platformName) {
+                case IOS:
+                    platformDefaults = (Map<String, Object>)defaults.get("ios");
+                    break;
+                case ANDROID:
+                    platformDefaults = (Map<String, Object>)defaults.get("android");
+                    break;
+            }
+            if(platformDefaults != null) {
+                readProperties(platformDefaults);
+            }
+        }
+
+        String environmentSchemes = System.getProperty("conductorCurrentSchemes");
+        if(environmentSchemes != null) {
+            currentSchemes = environmentSchemes.split(",");
+        } else {
+            List<String> schemesList = (List<String>)config.get("currentSchemes");
+            if(schemesList != null) {
+                currentSchemes = new String[schemesList.size()];
+                schemesList.toArray(currentSchemes);
+            }
+        }
+
+        if(currentSchemes != null) {
+
+            for(String scheme : currentSchemes) {
+                Map<String, Object> schemeData = (Map<String, Object>)config.get(scheme);
+                if(schemeData != null) {
+                    readProperties(schemeData);
+                }
+            }
+        }
+    }
+
+    private void readProperties(Map<String, Object> properties) {
+        for(String key : properties.keySet()) {
+            if(key.equals("ios") || key.equals("android")) {
+                // These keys are the start of platform specific options
+                // and should be ignored
+                continue;
+            }
+
+            setProperty(key, (String)properties.get(key).toString());
+        }
+    }
+
+    private void setProperty(String propertyName, String propertyValue) {
+        Method[] methods = this.getClass().getMethods();
+
+        String capPropertyKey = propertyName.substring(0, 1).toUpperCase() + propertyName.substring(1);
+        String methodName = "set" + capPropertyKey;
+        Method foundMethod = null;
+        for(Method method : methods) {
+            if(method.getName().equals(methodName)) {
```

Comment:
If you use `.equalsIgnoreCase( )` you can remove the string capitalization manipulation a few lines up.

Reply:
Technically, I'm pretty sure someone could be a jerk and create a differently cased method of the same name and it would be valid....